src/optimise/MCMCSamples.cpp

```diff
@@ -86,6 +96,31 @@ MCMCSamples::InitialiseHistograms(){
         f1DProjections = HistTools::MakeAllHists(histAxes, paramNames);
     }
 
+    if(fSaveChain){
+      stepClock.Start();
+      int parameterNumber = 0;
+      fChain = new TTree("posteriors", "Posterior_Distributions");
+      fChain->Branch("LogL", &fCurrentVal, "LogL/D");
+      fChain->Branch("Accepted", &fAccepted, "Accepted/O");
+      fChain->Branch("Step", &fStepNumber, "Step/I");
+      fChain->Branch("StepTime", &fStepTime, "StepTime/D");
+
+      parvals.reserve(paramNames.size());
+      for(std::set<std::string>::iterator it =  paramNames.begin();
+	  it != paramNames.end(); ++it){
+	//Strip out "-" from parameter names
+	const std::string& tempname = *it;
+	std::string bname = tempname; 
+	if(bname.find("-") != std::string::npos)
+          bname.replace(bname.find("-"), 1, "_");
+        char bit[40] = "";
```

Comment:
Can we assume that the length of `bname` concatenated with `"\D"` will always be less than 40 characters?
Unless I'm mistaken, you can just use `+` for string concatenation, and avoid needing to create a char* entirely - instead you can probably just do `const std::string bit = bname + "\D";`, and then do `bit.c_str()` on the `Branch()` line.

Reply:
Thanks for looking at this so quickly! You're definitely right about the branch names, I've fixed it to be as you suggest. I've made all the tabs 4 spaces now, sorry that was a classic emacs faux pas. And I've now added fSaveChain defaulting to false in both MCMC and MCMCSamples as we discussed in the meeting

Thanks a lot!